core/creator/src/main/java/io/quarkus/creator/phase/runnerjar/RunnerJarPhase.java

```diff
@@ -426,6 +431,55 @@ private void filterZipFile(Path resolvedDep, Path targetPath, Set<String> transf
         }
     }
 
+    /**
+     * Handles the core attributes and exception handling for the Manifest.
+     * Setting the Version, Main Class and Classpath
+     *
+     * @param attributes - Manifest Attributes
+     * @param classPath - The Main Class
+     * @throws IOException
+     */
+    private void handleCoreAttributes(final Attributes attributes, final String classPath) {
+        attributes.put(Attributes.Name.MANIFEST_VERSION, "1.0");
+        if (attributes.containsKey(Attributes.Name.CLASS_PATH)) {
+            log.warn(
+                    "Your MANIFEST.MF already defined a CLASS_PATH entry. Quarkus has overwritten this existing entry.");
+        }
+        attributes.put(Attributes.Name.CLASS_PATH, classPath);
+        if (attributes.containsKey(Attributes.Name.MAIN_CLASS)) {
+            String existingMainClass = attributes.getValue(Attributes.Name.MAIN_CLASS);
+            if (!mainClass.equals(existingMainClass)) {
+                log.warn("Your MANIFEST.MF already defined a MAIN_CLASS entry. Quarkus has overwritten your existing entry.");
+            }
+        }
+        attributes.put(Attributes.Name.MAIN_CLASS, mainClass);
+    }
+
+    /**
+     * Resolve any attribute name that is a place holder between ${...} notation.
+     * This is used by build systems that inject metadata to the manifest at build time,
+     * and generally created by CI/CD systems.
+     *
+     * @param attributes - Manifest Attributes to be resolved
+     */
+    private void handleResolvableAttributes(final Attributes attributes) {
+        attributes.keySet().forEach(k -> {
```

Comment:
Please just use loops for simple stuff like this. Lambda based iteration is way slower and uses a lot of metaspace that hangs around for the life of the program. This is not that much of an issue for deployment time code but the lambda does not even make the code less verbose in this case.